### ISSUE TITLE
saveRequestFiles hangs for some fields

### ISSUE THREAD

**Issue by vkurchatkin** [fastify/fastify-multipart#185]
##  🐛 Bug Report

I send a simple multipart request with just one file and consume them with  `req.saveRequestFiles()`. For some files it works, but for some other files never resolves. It seems that the code hangs here https://github.com/fastify/fastify-multipart/blob/5ca152453fc08a36f0ccd943af9f574db52526e7/index.js#L502 after successfully processing the file.

## To Reproduce

```js
const files = await req.saveRequestFiles();
```

## Expected behavior

Does not hang

## Your Environment

- *node version*: 14
- *fastify version*: 3.5.1
- *os*: Linux

**Comment by vkurchatkin** [comment]
It seems to be related to how `pipeline` interacts with `busboy`. Sometimes it triggers `onEnd`, sometimes it doesn't. If instead of `pump` we just resume the stream, `onEnd` is always triggered

**Comment by mcollina** [comment]
This looks _a lot_ like a bug in pipeline :( and node core. Let me also ping @ronag who worked on that lately.

Would you be able to start peeling the onion and get to a reproducible case _without_ the use of fastify or busboy?

**Comment by ronag** [comment]
Which Node 14.x version?

**Comment by vkurchatkin** [comment]
Here is the baseline repro (with fastify and fastify-multipart):

https://github.com/vkurchatkin/fastify-multipart-bug

I run it on Node 14.13.1

**Comment by ronag** [comment]
@vkurchatkin and what should I see when I run this?

**Comment by ronag** [comment]
@mcollina Why do you think this is a bug in pipeline?

@vkurchatkin Do you think you can find a repro using only busboy? I can't really see anything other than `onEnd` is not invoked for whatever reason. Busboy is implemented in a bit of a roundabout way so it's hard to follow what's happening.

**Comment by ronag** [comment]
The problem is that busboy for some reason emits `'close'` before `'finish'`. Probably since it goes around the usual streams framework and emits `'finish'` manually.

**Comment by ronag** [comment]
I think the problem here is that we changed `autoDestroy` default to `true`.

**Comment by mcollina** [comment]
I think we can just workaround it in here by setting `._writableState.autoDestroy = false` if it is true.
@vkurchatkin could you make a PR to this repo?

**Comment by victoradan** [comment]
Thank you for adding a fix for this. When is it planned to be released?

**Comment by Eomm** [comment]
Released in 3.3.1 👍
